Source/CronosPlayUnreal/Private/PlayCppSdkActor.cpp

```diff
@@ -408,3 +433,258 @@ void APlayCppSdkActor::destroyCoreClient() {
+                                         FCronosSignedTransactionDelegate Out) {
+
+  AsyncTask(ENamedThreads::AnyHiPriThreadNormalTask,
+            [this, Out, contractAddress, fromAddress, toAddress, amount,
+             gasLimit, gasPrice]() {
+              FString result;
+              FCronosSignedTransactionRaw txresult;
+              try {
+                if (NULL == _coreClient) {
+                  result = TEXT("Invalid Walletconnect");
+                } else {
+
+                  std::string mycontractaddress =
+                      TCHAR_TO_UTF8(*contractAddress);
+                  WalletConnectErc20Transfer info;
+                  info.contract_address = mycontractaddress;
+                  info.from_address = TCHAR_TO_UTF8(*fromAddress);
+                  info.to_address = TCHAR_TO_UTF8(*toAddress);
+                  info.amount = TCHAR_TO_UTF8(*amount);
+
+                  setCommon(info.common, fromAddress, gasLimit, gasPrice);
+
+                  copyVecToTArray(_coreClient->erc20_transfer(info),
+                                  txresult.SignedTx);
+                }
+
+              } catch (const rust::cxxbridge1::Error &e) {
+                result = FString::Printf(
+                    TEXT("CronosPlayUnreal Erc1155SafeTransferFrom Error: %s"),
+                    UTF8_TO_TCHAR(e.what()));
+              }
+
+              AsyncTask(ENamedThreads::GameThread, [Out, txresult, result]() {
+                Out.ExecuteIfBound(txresult, result);
+              });
+            });
+}
+
+void APlayCppSdkActor::setCommon(WalletConnectTxCommon &common,
+                                 FString fromaddress, FString gaslimit,
+                                 FString gasprice) {
+  std::string mycronosrpc = TCHAR_TO_UTF8(*myCronosRpc);
+  std::string myfromaddress = TCHAR_TO_UTF8(*fromaddress);
+  // get nonce
+  std::string mynonce =
+      org::defi_wallet_core::get_eth_nonce(myfromaddress.c_str(), mycronosrpc)
+          .c_str();
+
+  common.web3api_url = mycronosrpc.c_str();
+  common.chainid = myCronosChainID;
+  common.gas_limit = TCHAR_TO_UTF8(*gaslimit);
+  common.gas_price = TCHAR_TO_UTF8(*gasprice);
+  common.nonce = mynonce;
+}
+
+void APlayCppSdkActor::Erc20Approve(FString contractAddress,
+                                    FString fromAddress,
+                                    FString approvedAddress, FString amount,
+                                    FString gasLimit, FString gasPrice,
+                                    FCronosSignedTransactionDelegate Out) {
+
+  AsyncTask(ENamedThreads::AnyHiPriThreadNormalTask,
+            [this, Out, contractAddress, fromAddress, approvedAddress, amount,
+             gasLimit, gasPrice]() {
+              FString result;
+              FCronosSignedTransactionRaw txresult;
+              try {
+                if (NULL == _coreClient) {
+                  result = TEXT("Invalid Walletconnect");
+                } else {
```

Comment:
`result` is uninitialised  in else statement.

Reply:
default is "", i'll check